### DISCUSSION TITLE
docker image with theia ,java 17 and maven 3

### DISCUSSION THREAD

**Discussion by Jay-Bhatt98** [eclipse-theia/theia#12635] [Q&A]
i have created a docker image my base image is eclipse-temurin:17-jre-alpine after i have created a build when i run docker image in terminal i am able to find java version ,maven version and even my plugin were there as bulit in plugin but when i am creating a java file it says plugins not activating.i am not sure what i have done wrong is it a bug or do i have to add something else in my dockerfile and package .json
<img width="1784" alt="Screenshot 2023-06-21 at 5 10 12 PM" src="https://github.com/eclipse-theia/theia/assets/104554325/5b9a845d-e687-4807-b38e-66948bfe5adf">

[java17Dockerfile.txt](https://github.com/eclipse-theia/theia/files/11817062/java17Dockerfile.txt)
[package.json.txt](https://github.com/eclipse-theia/theia/files/11817065/package.json.txt)

**Comment by msujew** [comment]
I'm not too familiar with Java or the Java for vscode extension by Redhat, but one thing that stands out to me is this in your docker file:

```
ENV JAVA_HOME=/usr/local/openjdk-8
ENV PATH=$JAVA_HOME/bin:$PATH
```

Given that the extension doesn't complain that there's no java, but rather that it only supports Java starting with 11, I assume that it found the Java 8 version on the `PATH`.

If that's not the issue, you will be likely more successful trying to ask [them directly](https://github.com/redhat-developer/vscode-java/discussions) instead of us, as we are only implementing the vscode extension protocol, and have nothing do to with the extensions themselves.

**Reply by Jay-Bhatt98** [reply]
ENV JAVA_HOME /usr/lib/jvm/java-17-openjdk
even with openjdk 17 i am getting same 
[Uploading Dockerfile_old.txt…]()
